web/modules/custom/dpl_opening_hours/src/Model/OpeningHoursRepository.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+namespace Drupal\dpl_opening_hours\Model;
+
+use Drupal\Core\Database\Connection;
+use Drupal\node\NodeInterface;
+use Drupal\node\NodeStorageInterface;
+use Drupal\taxonomy\TermInterface;
+use Drupal\taxonomy\TermStorageInterface;
+use Psr\Log\LoggerInterface;
+use Safe\DateTimeImmutable;
+
+/**
+ * Repository for managing persistence of opening hours instance value objects.
+ */
+class OpeningHoursRepository {
+
+  const DATABASE_TABLE = 'dpl_opening_hours_instance';
+
+  /**
+   * Constructor.
+   */
+  public function __construct(
+    private LoggerInterface $logger,
+    private Connection $connection,
+    private NodeStorageInterface $branchStorage,
+    private TermStorageInterface $categoryTermStorage,
+  ) {}
+
+  /**
+   * Load a single opening hours instance.
+   */
+  public function load(int $id): ?OpeningHoursInstance {
+    $result = $this->connection->select(self::DATABASE_TABLE, self::DATABASE_TABLE)
+      ->fields(self::DATABASE_TABLE)
+      ->condition('id', $id)
+      ->execute();
+    if (!$result) {
+      return NULL;
+    }
+    $data = $result->fetchAssoc();
+    if (!is_array($data)) {
+      return NULL;
+    }
+
+    try {
+      return $this->toObject($data);
+    }
+    catch (\OutOfBoundsException $e) {
+      $this->logger->error("Unable to load opening hours instance: %message", ["%message" => $e->getMessage()]);
+      return NULL;
+    }
+  }
+
+  /**
+   * Load a collection of opening hours.
+   *
+   * @return OpeningHoursInstance[]
+   *   Opening hours instances which match the provided criteria.
+   */
+  public function loadMultiple(int $branchId = NULL, \DateTimeInterface $fromDate = NULL, \DateTimeInterface $toDate = NULL): array {
+    $query = $this->connection->select(self::DATABASE_TABLE, self::DATABASE_TABLE)
+      ->fields(self::DATABASE_TABLE);
+    if ($branchId) {
+      $query->condition('branch_nid', $branchId);
+    }
+    if ($fromDate) {
+      $query->condition('date', $fromDate->format('Y-m-d'), '>=');
+    }
+    if ($toDate) {
+      $query->condition('date', $toDate->format('Y-m-d'), '<=');
+    }
+
+    $result = $query->execute();
+    if (!$result) {
+      return [];
+    }
+
+    $maybe_objects = array_map(function (array $data): ?OpeningHoursInstance {
```

Comment:
"maybe objects"?

Reply:
Suggestions, choose your own adventure:

1. Rename it to "possible_objects", along with a code comment on why it's named that

OR (my preference):

2. filter out non-objects as you suggest, but maybe do it on two lines instead of one :)